llmfoundry/models/hf/model_wrapper.py

```diff
@@ -24,29 +42,68 @@
 # HuggingFace hardcodes the ignore index to -100
 _HF_IGNORE_INDEX = -100
 
+log = logging.getLogger(__name__)
 
 class HuggingFaceModelWithFSDP(HuggingFaceModel):
```

Comment:
to avoid this being a backwards compatibility breaking change, how about we make a new base class for everything to inherit from and deprecate HuggingFaceModelWithFSDP (which is not a very good name for the new class anyway)

Reply:
here is the warning we raise for deprecations: https://github.com/mosaicml/llm-foundry/blob/d3a9ac9b874442657fd0df48afae5de6414e9a2f/llmfoundry/utils/warnings.py#L15